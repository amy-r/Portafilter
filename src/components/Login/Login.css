.top-image{
  background: url(/header-triangle_1.png);
  height: 200px;
}