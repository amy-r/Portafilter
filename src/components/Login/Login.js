import React, { Component } from 'react';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { connect } from 'react-redux';
import { uiConfig } from '../../Utilities/firebase-config';
import { logIn, logOut } from '../../actions/index';
import * as firebase from 'firebase';
import { writeUserData, pullRoasters } from '../../Utilities/firebaseFunctions'
import { firebaseApp } from '../../Utilities/firebaseFunctions'

var db = firebaseApp.database();
var ref = db.ref('/roasters');

export class Login extends Component {
  constructor(props) {
    super(props)
  }

 async componentDidMount() {
   firebase.auth().onAuthStateChanged( user => {
      try {
        const userToStore = {
          userName: user.displayName,
          userEmail: user.email,
          userPhoto: user.photoUrl,
          userId: user.uid 
        }
        this.props.logIn(userToStore)
        writeUserData(userToStore)
        pullRoasters(ref)
      } catch (error) {
        console.log(error)
      }
    })
  }

  signOut = (user) => {
    firebase.auth().signOut()
    this.props.logOut(user)
  }

  render() {
    const { user, signOut } = this.props 
    if(!user.userName) {
      return(
        <div>
          <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} />
        </div>
      ) 
    } else {
       return (
        <div>
          <h1> Welcome, {user.userName} </h1>
          <img src={user.userPhoto} />
          <button onClick={this.signOut} > Sign Out</button>
        </div>
      )
    }
  }
}


export const mapStateToProps = store => ({
  user: store.user
})

export const mapDispatchToProps = dispatch => ({
  logIn: user => dispatch(logIn(user)),
  logOut: user => dispatch(logOut(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Login)
