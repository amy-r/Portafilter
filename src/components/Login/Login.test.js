import React from 'react';
import ReactDOM from 'react-dom';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { shallow } from 'enzyme';

describe('Login', () => {
  it('should exist', () => {
    const mockUser = {
      userName: 'Jordan',
      userEmail: 'Jordan@Project.com',
      userPhoto: 'photo-url',
      userId: 'fancy encryption'
    }

    const wrapper = shallow(<Login user={mockUser}/>, {disableLifecycleMethods:true});
    
    expect(wrapper).toMatchSnapshot();
  })
})

describe('MSTP and MDTP', () => {
  it('should define user props for the container', () => {
    const userName= "David Bowie";
    const userEmail= "david@gmail.com";
    const mockStore = {user: [ {name, userEmail} ]};
    const expected = [ {name, userEmail} ];
    const mapped = mapStateToProps(mockStore);
    expect(mapped.user).toEqual(expected)
  })

  it('should call the dispatch function on MDTP for logIn', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.logIn();
    expect(mockDispatch).toHaveBeenCalled;
  })

  it('should call the dispatch function on MDTP for logOut', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.logOut();
    expect(mockDispatch).toHaveBeenCalled;
  })

  it('should call the dispatch function on MDTP for retrievedRoasters', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.retrievedRoasters();
    expect(mockDispatch).toHaveBeenCalled;
  })
})