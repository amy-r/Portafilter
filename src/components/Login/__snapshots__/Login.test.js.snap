// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly with a user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    logIn={[Function]}
    logOut={[Function]}
    user={
        Object {
            "userEmail": "Jordan@Project.com",
            "userId": "fancy encryption",
            "userName": "Jordan",
            "userPhoto": "photo-url",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h1>
                 WELCOME
        </h1>
        <h3
                className="name"
        >
                Jordan
                 
        </h3>
        <img
                alt="user"
                className="profile-picture"
                src="photo-url"
        />
        <button
                className="logout"
                onClick={[Function]}
        >
                 LOG OUT
        </button>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             WELCOME
</h1>,
          <h3
            className="name"
>
            Jordan
             
</h3>,
          <img
            alt="user"
            className="profile-picture"
            src="photo-url"
/>,
          <button
            className="logout"
            onClick={[Function]}
>
             LOG OUT
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " WELCOME",
          },
          "ref": null,
          "rendered": " WELCOME",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Jordan",
              " ",
            ],
            "className": "name",
          },
          "ref": null,
          "rendered": Array [
            "Jordan",
            " ",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "user",
            "className": "profile-picture",
            "src": "photo-url",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " LOG OUT",
            "className": "logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " LOG OUT",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1>
                     WELCOME
          </h1>
          <h3
                    className="name"
          >
                    Jordan
                     
          </h3>
          <img
                    alt="user"
                    className="profile-picture"
                    src="photo-url"
          />
          <button
                    className="logout"
                    onClick={[Function]}
          >
                     LOG OUT
          </button>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
               WELCOME
</h1>,
            <h3
              className="name"
>
              Jordan
               
</h3>,
            <img
              alt="user"
              className="profile-picture"
              src="photo-url"
/>,
            <button
              className="logout"
              onClick={[Function]}
>
               LOG OUT
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " WELCOME",
            },
            "ref": null,
            "rendered": " WELCOME",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Jordan",
                " ",
              ],
              "className": "name",
            },
            "ref": null,
            "rendered": Array [
              "Jordan",
              " ",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "user",
              "className": "profile-picture",
              "src": "photo-url",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " LOG OUT",
              "className": "logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": " LOG OUT",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Login should render correctly without a user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    logIn={[Function]}
    logOut={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="header-container"
        >
                <img
                        alt="Roastr Logo"
                        className="header-triagle"
                        src="header-triangle_2.svg"
                />
        </div>
        <h3>
                 welcome to 
        </h3>
        <img
                alt="Roastr logo"
                className="landing-logo"
                src="logo.svg"
        />
        <e
                firebaseAuth={
                        Object {
                                "apiKey": "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
                                "appName": "[DEFAULT]",
                                "authDomain": "roastr-e35c3.firebaseapp.com",
                                "currentUser": null,
                              }
                }
                uiConfig={
                        Object {
                                "callbacks": Object {
                                  "signInSuccess": [Function],
                                },
                                "signInFlow": "popup",
                                "signInOptions": Array [
                                  "google.com",
                                ],
                              }
                }
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-container"
>
            <img
                        alt="Roastr Logo"
                        className="header-triagle"
                        src="header-triangle_2.svg"
            />
</div>,
          <h3>
             welcome to 
</h3>,
          <img
            alt="Roastr logo"
            className="landing-logo"
            src="logo.svg"
/>,
          <e
            firebaseAuth={
                        Object {
                                    "apiKey": "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
                                    "appName": "[DEFAULT]",
                                    "authDomain": "roastr-e35c3.firebaseapp.com",
                                    "currentUser": null,
                                  }
            }
            uiConfig={
                        Object {
                                    "callbacks": Object {
                                      "signInSuccess": [Function],
                                    },
                                    "signInFlow": "popup",
                                    "signInOptions": Array [
                                      "google.com",
                                    ],
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Roastr Logo"
              className="header-triagle"
              src="header-triangle_2.svg"
/>,
            "className": "header-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Roastr Logo",
              "className": "header-triagle",
              "src": "header-triangle_2.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " welcome to ",
          },
          "ref": null,
          "rendered": " welcome to ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Roastr logo",
            "className": "landing-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "firebaseAuth": Object {
              "apiKey": "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
              "appName": "[DEFAULT]",
              "authDomain": "roastr-e35c3.firebaseapp.com",
              "currentUser": null,
            },
            "uiConfig": Object {
              "callbacks": Object {
                "signInSuccess": [Function],
              },
              "signInFlow": "popup",
              "signInOptions": Array [
                "google.com",
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="header-container"
          >
                    <img
                              alt="Roastr Logo"
                              className="header-triagle"
                              src="header-triangle_2.svg"
                    />
          </div>
          <h3>
                     welcome to 
          </h3>
          <img
                    alt="Roastr logo"
                    className="landing-logo"
                    src="logo.svg"
          />
          <e
                    firebaseAuth={
                              Object {
                                        "apiKey": "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
                                        "appName": "[DEFAULT]",
                                        "authDomain": "roastr-e35c3.firebaseapp.com",
                                        "currentUser": null,
                                      }
                    }
                    uiConfig={
                              Object {
                                        "callbacks": Object {
                                          "signInSuccess": [Function],
                                        },
                                        "signInFlow": "popup",
                                        "signInOptions": Array [
                                          "google.com",
                                        ],
                                      }
                    }
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header-container"
>
              <img
                            alt="Roastr Logo"
                            className="header-triagle"
                            src="header-triangle_2.svg"
              />
</div>,
            <h3>
               welcome to 
</h3>,
            <img
              alt="Roastr logo"
              className="landing-logo"
              src="logo.svg"
/>,
            <e
              firebaseAuth={
                            Object {
                                          "apiKey": "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
                                          "appName": "[DEFAULT]",
                                          "authDomain": "roastr-e35c3.firebaseapp.com",
                                          "currentUser": null,
                                        }
              }
              uiConfig={
                            Object {
                                          "callbacks": Object {
                                            "signInSuccess": [Function],
                                          },
                                          "signInFlow": "popup",
                                          "signInOptions": Array [
                                            "google.com",
                                          ],
                                        }
              }
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Roastr Logo"
                className="header-triagle"
                src="header-triangle_2.svg"
/>,
              "className": "header-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Roastr Logo",
                "className": "header-triagle",
                "src": "header-triangle_2.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " welcome to ",
            },
            "ref": null,
            "rendered": " welcome to ",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Roastr logo",
              "className": "landing-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "firebaseAuth": Object {
                "apiKey": "AIzaSyBYOjbvNWIvFDu-9R6gR4PcBqkYt5H98g4",
                "appName": "[DEFAULT]",
                "authDomain": "roastr-e35c3.firebaseapp.com",
                "currentUser": null,
              },
              "uiConfig": Object {
                "callbacks": Object {
                  "signInSuccess": [Function],
                },
                "signInFlow": "popup",
                "signInOptions": Array [
                  "google.com",
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
